src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -85,7 +85,18 @@
                                                                                                ((some-fn :db
                                                                                                          :dbname
                                                                                                          :sid
-                                                                                                         :catalog))))})
+                                                                                                         :catalog))))
+   "debugUnreturnedConnectionStackTraces" true        ; print a stack trace from when a leaked connection was acquired
+                                                      ; if it was never returned
+   ;; this will kill ANY connection that is still active 10 min after being checked out from the pool
+   ;; override with MB_C3P0_UNRETURNED_CONN_TIMEOUT (in seconds)
+   "unreturnedConnectionTimeout"          (if-let [v (System/getenv "MB_C3P0_UNRETURNED_CONN_TIMEOUT")]
```

Comment:
(note to team if we try to get this merged in) using `env/env` or the stuff in `metabase.config` would be preferable because you can set those with Java `-D` properties as well which means these could be set in your `~/.clojure/deps.edn` if you're running with the Clojure CLI